workflow/lint/lint.go

```diff
@@ -0,0 +1,240 @@
+package lint
```

Comment:
Is this code just called by the CLI? If so it shouldn't be under `workflow/`

Reply:
@simster7 ATM yes, it is called only from the CLI. Where do you think it should be instead?

Reply:
Somewhere under `cmd/argo`. Maybe `cmd/argo/lint`?